integration-tests/main/src/main/java/io/quarkus/it/metrics/DummyGauge.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.it.metrics;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.event.Observes;
+import javax.interceptor.AroundConstruct;
+
+import org.eclipse.microprofile.metrics.MetricUnits;
+import org.eclipse.microprofile.metrics.annotation.Gauge;
+
+import io.quarkus.runtime.StartupEvent;
+
+/**
+ * The purpose of this bean is to test that ArC automatically registers the constructor for reflection if an
+ * {@link AroundConstruct} interceptors is used. We use a dedicated bean to make sure the class is not registered by another
+ * extension, ie. by resteasy in case of JAX-RS resource.
+ * 
+ * See https://github.com/quarkusio/quarkus/issues/6898
+ */
+@ApplicationScoped
+public class DummyGauge {
```

Comment:
Shouldn't we add an assertion in `io.quarkus.it.main.MetricsTestCase` or somewhere to make sure the gauge really gets registered?

Reply:
Hm, we should. But we will have to add `@RegisterForReflection` too, due to https://github.com/quarkusio/quarkus/issues/6993. @jmartisk would you care to add this assertion to my PR? ;-)

Reply:
Oh, I thought the changes in this PR mean we don't need that annotation anymore. Ok.
How do I add something to your PR? I don't think I have permissions for that.

Reply:
Nope, this PR merely fixes the problem in ArC where we did not register the constructor for reflection if `@AroundConstruct` was used...

Reply:
@jmartisk as a member of the project you can push changes to any PR.

Reply:
But how? Which remote repo and branch should I push to? It won't allow me to push into Martin's fork.
Per https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork it seems it could work if I had push access to Quarkus repo, which I don't

Reply:
I am pretty sure you and I have the same push access and pushing to a PR works for me.
This is what I do:

* Add a git remote for the user's fork (using ssh, not https as the protocol) - for example `foo`
* Check out the PR in a branch, say `pull/1234`
* Make changes and commit
* Push like so: `git push foo pull/1234:name-of-branch-in-users-fork`

Reply:
Well, that's what I did and receive this on push
```
ERROR: Permission to mkouba/quarkus.git denied to jmartisk.
```
The document explicitly says that I can do this if the PR is opened against a repository where I have push access to - that's probably the issue.
Anyway I pushed it to my own branch https://github.com/jmartisk/quarkus/commit/7025a983a5aba54e5ba5c7bb6eba8d88199fe3e0 so I guess you guys can cherry-pick it

Reply:
Ok, will do. Apparently you don't have the permissions. Thanks!